src/libtexture/imagecache.cpp

```diff
@@ -467,6 +470,7 @@ ImageCacheFile::open (ImageCachePerThreadInfo *thread_info)
     // Since each subimage can potentially have its own mipmap levels,
     // keep track of the highest level discovered
```

Comment:
So subtracting the old and adding the new is to account for the case where you might be re-reading the same file?